tests/server/models/test_flow_run_states.py

```diff
@@ -49,14 +47,8 @@ async def test_locks_row_for_update(self, db, flow_run):
             - If not locked correctly, session 2 will see SCHEDULED -> PENDING
 
         """
-        connection_url = PREFECT_API_DATABASE_CONNECTION_URL.value()
-        dialect = get_dialect(connection_url)
-
-        if dialect.name == "sqlite":
-            pytest.skip("SQLite does not support row-level locking")
```

Comment:
When confirming this actually failed on SQLite before my changes it turns out that the way the events were set up ensured this test would pass even without the locking. I've updated the test to cover the actual regression.